frontend/src/components/TimetableGrid.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { X } from "lucide-react";
+import { useMemo, useState } from "react";
+
+export function TimetableGrid({
+  isVertical,
+  timetableDetailsSections,
+  handleUnitClick,
+  handleUnitDelete,
+}: {
+  isVertical: boolean;
+  timetableDetailsSections: {
+    id: string;
+    name: string;
+    roomTime: string[];
+    courseId: string;
+    type: string;
+    number: number;
+    instructors: string[];
+  }[];
+  handleUnitClick: (
+    e: {
+      id: string;
+      name: string;
+      courseId: string;
+      room: string;
+      code: string;
+      type: string;
+      number: number;
+      instructors: string[];
+    } | null,
+    event: React.MouseEvent,
+  ) => void;
+  handleUnitDelete: (
+    e: {
+      id: string;
+      name: string;
+      courseId: string;
+      room: string;
+      code: string;
+      type: string;
+      number: number;
+      instructors: string[];
+    } | null,
+  ) => void;
+}) {
+  const daysOfWeek = ["M", "T", "W", "Th", "F", "S"];
+
+  const timings = [
+    "8 - 9",
+    "9 - 10",
+    "10 - 11",
+    "11 - 12",
+    "12 - 1",
+    "1 - 2",
+    "2 - 3",
+    "3 - 4",
+    "4 - 5",
+    "5 - 6",
+    "6 - 7",
+  ];
+
+  const [displayRows, setDisplayRows] = useState(6);
```

Comment:
```suggestion
  const displayRows = 6;
```
Since this is not changed anywhere, you can define it as constant.